app/views/data_view.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+from app.models import Donor, Donation, Item
+from django.http import HttpResponse, JsonResponse, HttpResponseBadRequest
+from django.views import View
+import simplejson as json
+
+# return aggregate value for given time interval
+# if request.GET['interval'] is false, return value of all items
+@login_required(login_url='/login')
+def aggregate_value(request):
+    return_value = 0
+    if request.GET['interval'] is False:
+        return_value = sum(Item.value for item in Item.objects.all())
+    else:
+        initial_date = request.GET['initial_date']
+        final_date = request.GET['final_date']
+        items = Item.objects.filter(
+            tax_receipt_no__donate_date__gte=initial_date
```

Comment:
Rather than filtering twice. You can use the __range to filter. The donate format has to be "YYYY-MM-DD".

ie) `Donation.objects.filter(donate_date__range=["2016-01-01", "2018-01-01"])`